File: Data Structures & Algorithms/daily-temperatures/submission-1.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        result = [0] * len(temperatures)
        stack = []

        for i in range(len(temperatures)):
            while stack and temperatures[i] > temperatures[stack[-1]]:
                day_idx = stack.pop()
                result[day_idx] = i - day_idx

            stack.append(i)
        
        return result

File: Data Structures & Algorithms/count-substrings-with-only-one-distinct-letter/submission-1.cpp
class Solution {
public:
    int countLetters(string s) {
        int start = 0;
        int result = 0;
        for (int i = 1; i <= s.size(); i++) {
            if (i == s.size() || s[i] != s[start]) {
                result += (i-start) * (i-start+1)/2;
                start = i;
            }
        }
        return result;
    }
};
